src/utils/UrlUtils.ts

```diff
@@ -104,6 +104,9 @@ export class UrlUtils {
         if (Utils.isNullOrUndefined(value) || Utils.isNullOrUndefined(key)) {
           return '';
         }
+        if (isString(value) && Utils.isEmptyString(value)) {
```

Comment:
Two questions related to this `if`:

1- What happens if it is not there and we get an empty string? Is it needed or more of an optimization?
2- If needed, what about adding it to the `if` on line 104 (and simplifying to a "not truthy" check)?

```
if (Utils.isNullOrUndefined(value) || Utils.isEmptyString(value) || Utils.isNullOrUndefined(key)) {
     return '';
}

or assuming "value" cannot be a number, just

if (!value || Utils.isNullOrUndefined(key)) {
     return '';
}
```

Reply:
Why does the `isNonEmptyString` checks if type is string but `isEmptyString` doesn't do so?  `isEmptyString` would in principle return `true` on any other type than `string`

Reply:
Fair point, In this case I can simplify this to a falsy check and cover all cases correctly.

For the isEmptyString vs isNonEmptyString, it's a bit weird indeed, but if you look at the code of `Utils.ts`, you will see why : `isEmptyString(1234) -> true` 🤦‍♂️ 

So it should probably be changed, but im a bit afraid to change it since quite a bit of client  implementation are relying on those utils calls (even if they are not documented), and those are "old" classes that haven't changed in a very long time.

So I prefer to bite the bullet, chicken out, and leave it as is.